### PR TITLE
Fixed bug: SMAP JPL 

### PR DESCRIPTION
**Pull request by ShastriPaturi**
## Description

This PR fixes the bug reported in #493 
The SMAP SSS converter works for JPL h5 and RSS nc L2 SSS feeds.
Also updates the smos_sss2ioda.py with @SCRIPT_LIB_PATH@ version

[See the JEDI Documentation](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/inside/practices/pullrequest.html) for further information.

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #<493>

### CODE REVIEW

**Comment by ShastriPaturi** [comment]
@aerorahul, @mjkagnes123 and @HamidehGMAO ,

The commented lines: 123-127 is the QC flag from RTOFS-DA. I can uncomment that and add it to the converter or can delete it.

File: src/marine/smap_sss2ioda.py [hunk 1]
```diff
@@ -17,7 +16,7 @@
 import dateutil.parser
 from pathlib import Path
 
-IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+IODA_CONV_PATH = Path(__file__).parent/"../lib/pyiodaconv"
```
Comment (srherbener) [inline src/marine/smap_sss2ioda.py:19]: The other scripts have the deleted version of this line (ie the one using `@SCRIPT_LIB_PATH@`). Shouldn't this be the case here as well?

**Comment by mjkagnes123** [comment]
@ShastriPaturi I tested the script and it worked. I could process JPL SMAP data successfully! Thank you for fixing the script. Good work!

**Review by mjkagnes123 — APPROVED** [review]
I tested the script and it worked. I could process JPL SMAP data successfully! Thank you for fixing the script. Good work!

File: src/marine/smap_sss2ioda.py (hunk 1 above)
Comment (ShastriPaturi) [reply src/marine/smap_sss2ioda.py:19]: in smos_sss2ioda.py, the 
IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
does not exist, so this converter too does not have it.
Let me know.

Comment (srherbener) [reply src/marine/smap_sss2ioda.py:19]: I think the `@SCRIPT_LIB_PATH@` version is needed to get the `make install` version of the script to work correctly. However, I'm not certain. @aerorahul or @CoryMartin-NOAA could you confirm this? Thanks!

Comment (CoryMartin-NOAA) [reply src/marine/smap_sss2ioda.py:19]: Yes that is my understanding, because it may not be stored in ../lib/pyiodaconv

Comment (ShastriPaturi) [reply src/marine/smap_sss2ioda.py:19]: @srherbener, I will add that. However, I will have to add to the smos_sss2ioda.py as well.

Comment (srherbener) [reply src/marine/smap_sss2ioda.py:19]: @ShastriPaturi that sounds great, and thanks for taking care of this as well as the smos script.

**Review by srherbener — APPROVED** [review]
Thanks for putting the `@SCRIPT_LIB_PATH@` code back in.

Are you going to include the smos script with this PR, or do a subsequent PR?

Thanks!

**Review by ShastriPaturi — COMMENTED** [review]
> Thanks for putting the `@SCRIPT_LIB_PATH@` code back in.
> 
> Are you going to include the smos script with this PR, or do a subsequent PR?
> 
> Thanks!

I will add to this PR,

**Comment by ShastriPaturi** [comment]
> > Thanks for putting the `@SCRIPT_LIB_PATH@` code back in.
> > Are you going to include the smos script with this PR, or do a subsequent PR?
> > Thanks!
> 
> I will add to this PR,

@srherbener, I updated the smos_sss2ioda.py with @SCRIPT_LIB_PATH@ version

**Comment by srherbener** [comment]
@ShastriPaturi thanks for updating the smos script. There are some coding norms errors and a syntax error in the smos script (I think due to a missing double quote mark).

```
test 5
--
936 | Start  5: iodaconv_marine_coding_norms
937 |  
938 | 5: Test command: /build_container/bin/iodaconv_lint.sh "/jcsda/ioda-bundle/iodaconv/src/marine" "/jcsda/ioda-bundle/iodaconv"
939 | 5: Environment variables:
940 | 5:  OMP_NUM_THREADS=1
941 | 5: Test timeout computed to be: 10000000
942 | 5: + SRC_DIR=/jcsda/ioda-bundle/iodaconv/src/marine
943 | 5: + PYCODESTYLE_CFG_PATH=/jcsda/ioda-bundle/iodaconv
944 | 5: + cd /jcsda/ioda-bundle/iodaconv/src/marine
945 | 5: + pycodestyle -v --config=/jcsda/ioda-bundle/iodaconv/.pycodestyle --filename=*.py,*.py.in .
946 | 5: cli configuration: /jcsda/ioda-bundle/iodaconv/.pycodestyle
947 | 5: directory .
948 | 5: checking ./argoClim2ioda.py
949 | 5: checking ./cryosat_ice2ioda.py
950 | 5: checking ./emc_ice2ioda.py
951 | 5: checking ./gds2_sst2ioda.py
952 | 5: checking ./glider2ioda.py
953 | 5: checking ./gmao_obs2ioda.py
954 | 5: checking ./godae_profile2ioda.py
955 | 5: checking ./godae_ship2ioda.py
956 | 5: checking ./godae_trak2ioda.py
957 | 5: checking ./hgodas_adt2ioda.py
958 | 5: checking ./hgodas_insitu2ioda.py
959 | 5: checking ./hgodas_sst2ioda.py
960 | 5: checking ./ndbc_hfradar2ioda.py
961 | 5: checking ./rads_adt2ioda.py
962 | 5: checking ./smap_sss2ioda.py
963 | 5: checking ./smos_sss2ioda.py
964 | 5: ./smos_sss2ioda.py:17:39: E225 missing whitespace around operator
965 | 5: checking ./viirs_modis_l2_oc2ioda.py
966 | 5: checking ./viirs_modis_l3_oc2ioda.py
967 | 5: 1       E225 missing whitespace around operator
968 | 1/2 Test  #5: iodaconv_marine_coding_norms .....***Failed    0.36 sec
969 | test 26
970 | Start 26: test_iodaconv_smos_sss
971 |  
972 | 26: Test command: /build_container/bin/iodaconv_comp.sh "netcdf" "/build_container/bin/smos_sss2ioda.py
973 | -i testinput/SM_REPR_MIR_OSUDP2_20100601T000131_20100601T001849_662_120_1.nc
974 | -o testrun/smos_sss_l2.nc
975 | -d 2010060100" "smos_sss_l2.nc" "0.0"
976 | 26: Environment variables:
977 | 26:  OMP_NUM_THREADS=1
978 | 26: Test timeout computed to be: 10000000
979 | 26:   File "/build_container/bin/smos_sss2ioda.py", line 17
980 | 26:     IODA_CONV_PATH = Path(__file__).parent/"../lib/pyiodaconv
981 | 26:                                                             ^
982 | 26: SyntaxError: EOL while scanning string literal
983 | 2/2 Test #26: test_iodaconv_smos_sss ...........***Failed    0.02 sec
984 |  
985 | 0% tests passed, 2 tests failed out of 2
986 |  
987 | Label Time Summary:
988 | iodaconv    =   0.38 sec*proc (2 tests)
989 | script      =   0.38 sec*proc (2 tests)
990 |  
991 | Total Test time (real) =   0.39 sec
992 |  
993 | The following tests FAILED:
994 | 5 - iodaconv_marine_coding_norms (Failed)
995 | 26 - test_iodaconv_smos_sss (Failed)
996 | Errors while running CTest
```

File: src/marine/smos_sss2ioda.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 import netCDF4 as nc
 from pathlib import Path
 
-IODA_CONV_PATH = Path(__file__).parent/"../lib/pyiodaconv"
+IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@
```
Comment (srherbener) [inline src/marine/smos_sss2ioda.py]: Missing a double quote at the end of the line.

**Review by srherbener — CHANGES_REQUESTED** [review]

File: src/marine/smos_sss2ioda.py (hunk 2 above)
Comment (srherbener) [reply src/marine/smos_sss2ioda.py]: Sorry, I sent this at the same time you pushed the fix. Thanks for taking care of this!

**Review by srherbener — APPROVED** [review]

File: src/marine/smos_sss2ioda.py (hunk 2 above)
Comment (ShastriPaturi) [reply src/marine/smos_sss2ioda.py]: @srherbener no issues.
Please check now.

**Pull request merged**